From f53e3bf7239cacd57aaae487a79075877ae3af60 Mon Sep 17 00:00:00 2001
From: Stefan Wahren <stefan.wahren@chargebyte.com>
Date: Fri, 5 May 2023 07:54:38 +0200
Subject: [PATCH] src/bundle: allow exFAT as local file system

At least since Linux 5.7 the Kernel supports the exFAT file system. So
allow it to use modern flash drives as local file system in rauc.

Signed-off-by: Stefan Wahren <stefan.wahren@chargebyte.com>
---
 src/bundle.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/bundle.c b/src/bundle.c
index 6adfd0e..b03b5f2 100644
--- a/src/bundle.c
+++ b/src/bundle.c
@@ -33,6 +33,9 @@
 #ifndef CRAMFS_MAGIC
 #define CRAMFS_MAGIC 0x28cd3d45
 #endif
+#ifndef EXFAT_SUPER_MAGIC
+#define EXFAT_SUPER_MAGIC 0x2011bab0
+#endif
 #ifndef EXT4_SUPER_MAGIC /* also covers ext2/3 */
 #define EXT4_SUPER_MAGIC 0xef53
 #endif
@@ -1490,6 +1493,7 @@ static gboolean check_bundle_access(int bundle_fd, GError **error)
 			case AFS_SUPER_MAGIC:
 			case BTRFS_SUPER_MAGIC:
 			case CRAMFS_MAGIC:
+			case EXFAT_SUPER_MAGIC:
 			case EXT4_SUPER_MAGIC: /* also covers ext2/3 */
 			case F2FS_SUPER_MAGIC:
 			case ISOFS_SUPER_MAGIC:
-- 
2.34.1

